app/src/main/java/com/anytypeio/anytype/ui/sharing/SharingFragment.kt

```diff
@@ -15,23 +15,42 @@ import androidx.lifecycle.compose.collectAsStateWithLifecycle
 import androidx.navigation.fragment.findNavController
 import com.anytypeio.anytype.R
 import com.anytypeio.anytype.core_utils.ext.arg
+import com.anytypeio.anytype.core_utils.ext.argStringList
 import com.anytypeio.anytype.core_utils.ext.toast
 import com.anytypeio.anytype.core_utils.ui.BaseBottomSheetComposeFragment
 import com.anytypeio.anytype.di.common.componentManager
 import com.anytypeio.anytype.presentation.home.OpenObjectNavigation
 import com.anytypeio.anytype.presentation.sharing.AddToAnytypeViewModel
 import com.anytypeio.anytype.ui.editor.EditorFragment
 import com.anytypeio.anytype.ui.settings.typography
+import java.lang.IllegalStateException
 import javax.inject.Inject
 
 class SharingFragment : BaseBottomSheetComposeFragment() {
 
     private val sharedData : SharingData get() {
-        val result = arg<String>(SHARING_DATE_KEY)
-        return if (URLUtil.isValidUrl(result)) {
-            SharingData.Url(result)
+        val args = requireArguments()
+        return if (args.containsKey(SHARING_TEXT_KEY)) {
+            val result = arg<String>(SHARING_TEXT_KEY)
+            if (URLUtil.isValidUrl(result)) {
+                SharingData.Url(result)
+            } else {
+                SharingData.Raw(result)
+            }
+        } else if (args.containsKey(SHARING_IMAGE_KEY)) {
+            val result = arg<String>(SHARING_IMAGE_KEY)
+            SharingData.Image(uri = result)
+        } else if (args.containsKey(SHARING_FILE_KEY)) {
+            val result = arg<String>(SHARING_FILE_KEY)
+            SharingData.File(uri = result)
+        }else if (args.containsKey(SHARING_MULTIPLE_IMAGES_KEY)) {
+            val result = argStringList(SHARING_MULTIPLE_IMAGES_KEY)
+            SharingData.Images(uris = result)
+        } else if (args.containsKey(SHARING_MULTIPLE_FILES_KEY)) {
+            val result = argStringList(SHARING_MULTIPLE_FILES_KEY)
+            SharingData.Files(uris = result)
         } else {
-            SharingData.Raw(result)
+            throw IllegalStateException("Unexpected result")
```

Comment:
I think it would be better to log an error using Timber.e at this point and display a toast message to the user.

Reply:
currently it's not possible to change this code quickly, i'll see what I can do in the PR, thanks